### PR TITLE
vendor: github.com/moby/swarmkit/v2 v2.0.0-20220721174824-48dd89375d0a, change "csi" to "cluster"

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff: https://github.com/moby/swarmkit/compare/6068d1894d46...48dd89375d0a3e8a333c52bce1e6297e0b192d06

- change CSI to Cluster in API (https://github.com/moby/swarmkit/pull/3067)

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -214,12 +214,14 @@ definitions:
           - `volume` a docker volume with the given `Name`.
           - `tmpfs` a `tmpfs`.
           - `npipe` a named pipe from the host into the container.
+          - `cluster` TODO add description
```
Comment (thaJeztah) [inline api/swagger.yaml:217]: Not sure exactly where the `Type` is used, and where `cluster` / `csi` can be used (do we check for incorrect types for these?)

File: api/types/mount/mount.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ const (
 	// TypeNamedPipe is the type for mounting Windows named pipes
 	TypeNamedPipe Type = "npipe"
 	// TypeCluster is the type for Swarm Cluster Volumes.
-	TypeCluster = "csi"
+	TypeCluster Type = "cluster"
```
Comment (thaJeztah) [inline api/types/mount/mount.go:21]: I _think_ the intent of @dperny was to change this from `csi` to `cluster` ?


There's other occurrences of CSI in the codebase though, and I wasn't sure which ones should be renamed and which ones not.

File: api/types/volume/cluster_volume.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ type AccessMode struct {
 	BlockVolume *TypeBlock `json:",omitempty"`
 }
 
-// Scope defines the Scope of a CSI Volume. This is how many nodes a
+// Scope defines the Scope of a Cluster Volume. This is how many nodes a
```
Comment (thaJeztah) [inline api/types/volume/cluster_volume.go:107]: e.g.; these; wasn't sure where we wanted to use the "Cluster" terminology and where we would still have to refer to "CSI"

**Comment by dperny** [comment]
@thaJeztah The only real changes to make were in documentation. Everything else is appropriate.

The remaining references to "CSI" in the code base, especially in the plugins code, are appropriate, because they accurately reflect information about a CSI plugin, which is used to power a Cluster Volume.

This is the diff of my changes. If you'd like, I can just open my own PR to supersede this one, or you can pull these few changes into this one.

https://github.com/moby/moby/compare/master...dperny:docker:change_csi_cluster_string

**Comment by thaJeztah** [comment]
Thanks @dperny - updated with your changes 👍

**Comment by thaJeztah** [comment]
@corhere @cpuguy83 @tianon PTAL

**Comment by thaJeztah** [comment]
Also prepared a cherry-pick; https://github.com/moby/moby/pull/43938

**Comment by dperny** [comment]
LGTM

**Review by corhere — APPROVED** [review]

**Comment by thaJeztah** [comment]
I'll take @dperny's review; let's get this in (and I'll move https://github.com/moby/moby/pull/43938 out of draft)

**Pull request merged**